test/findall.jl

```diff
@@ -0,0 +1,87 @@
+# Helper function to test that findall(in(interval), x) works. By
+# default, a reference is generated using the general algorithm,
+# linear in complexity, by generating a vector with the same contents
+# as x.
+function assert_in_interval(x, interval,
+                            reference=findall(v -> v ∈ interval, collect(x)))
+    res = findall(in(interval), x)
+    @test res == reference || isempty(res) && isempty(reference)
+end
+
+@testset "Interval coverage" begin
+    x = range(0, stop=1, length=21)
+
+    @testset "Two intervals" begin
+        assert_in_interval(x, 0..0.5, 1:11)
+        assert_in_interval(x, 0..0.5)
+        assert_in_interval(x, Interval{:closed,:open}(0,0.5), 1:10)
+        assert_in_interval(x, Interval{:closed,:open}(0,0.5))
+        assert_in_interval(x, Interval{:closed,:open}(0.25,0.5), 6:10)
+        assert_in_interval(x, Interval{:closed,:open}(0.25,0.5))
+    end
+
+    @testset "Three intervals" begin
+        assert_in_interval(x, Interval{:closed,:open}(0,1/3), 1:7)
+        assert_in_interval(x, Interval{:closed,:open}(1/3,2/3), 8:14)
+        assert_in_interval(x, 2/3..1, 15:21)
+    end
+
+    @testset "Open interval" begin
+        assert_in_interval(x, OpenInterval(0.2,0.4), 6:8)
+    end
+
+    @testset "Random intervals" begin
+        @testset "L=$L" for L=[:closed,:open]
+            @testset "R=$R" for R=[:closed,:open]
+                for i = 1:1 # 20
+                    interval = Interval{L,R}(minmax(rand(),rand())...)
```

Comment:
Is the random number seeded somewhere? e.g. `Random.seed!(0)`

Reply:
No, it is not. In fact, I am not sure what best practice is?

Reply:
It's generally a good idea to seed the random number generator so that the tests are deterministic: otherwise, degenerate choices of floating point numbers may break the tests on other machines.

You can do this by adding Random to the test dependencies in (Project.toml), and do
```julia
using Random
Random.seed!(0)
```
where `0` can be an arbitrary number.

Reply:
👍